src/main/io/vtx_smartaudio.c

```diff
@@ -762,6 +739,14 @@ bool vtxSmartAudioInit(void)
     vtxSmartAudio.powerValues = saSupportedPowerValues;
 #endif
 
+    //This little trick saves us from storing the entire default frequency table just to find out if the frequencies are default or not.
+    //The default frequency table has a crc of 0x40. So if the current frequency table matches that, they are default.
+    uint8_t freqTableCRC = CRC8((uint8_t *)vtxSmartAudio.frequencyTable, sizeof(uint16_t) * vtxSmartAudio.capability.bandCount * vtxSmartAudio.capability.channelCount);
```

Comment:
While I like this approach for its simplicity, I am not sure it is the right one - one of the main intentions behind VTX tables is to allow users to define custom frequency tables that restrict them to the frequencies that they are legally allowed to use in the jurisdiction they are flying in. In most places around the globe this will be a subset of the frequencies supported by the VTX hardware. If we do not want to encourage users to instead use the full frequency list supported by their VTX (and in the process use Betaflight to break the law), we should also support channel mode if the channels defined in the VTX table are all available as discrete channels in the VTX hardware - even if this means storing a full list - I don't think there is a cheaper way to achieve what we want.

Reply:
Well if the user wants to use the full frequency table he will still have to copy all frequencies into vtxTable. The CRC check just finds out if all the entered frequencies match the default table or not. An empty vtxTable will still prevent the vtx from working at all.

Reply:
Yes - but pretty much no user will be able to use the full frequency table (that is in your proposal the only way to get channel mode to work) in a legal way, so adding this as it is will only serve to entice users to break the law, and should not be done.

Reply:
Ah you want a way to enable channel mode but with fewer frequencies. Sounds like a good idea.

We just have to deal with holes in the table properly. Lets take raceband for example. Suppose the local laws only allow the center four channels. In frequency mode, vtxTable could be set up like this:
`vtxtable band 5 RACEBAND R 5732 5769 5806 5843`
However, in index mode that would cause the vtx to go into channels 1-4, not 2-6 like we wanted.
So maybe this could be solved by introducing an empty placeholder like 0 or -1.
`vtxtable band 5 RACEBAND R 0 0 5732 5769 5806 5843 0 0`

Reply:
> Ah you want a way to enable channel mode but with fewer frequencies.

Yep. Sorry for not making this clear the first time around.

The approach with a placeholder is definitely a way to get there. I am not sure that we even have to enforce that frequencies are in the 'correct' slot in a band in the VTX table - I see allowing users to create one or more 'mix and match' bands of all of the frequencies they want to use / are allowed to use in the VTX table as a feature. We could either do this by verifying that all of the frequencies are mappable on init, or even in a way that tried to map, and if possible used a discrete channel if available, or direct frequency entry otherwise, every time a new frequency was selected.
Both of these have the drawback that in the few cases where the same frequency is defined by more than one band it will not be possible to distinguish the bands - but I suspect this problem will occur for pretty much any approach that is less strict than the currently proposed 'full match' - even if we only allow empty placeholders (including entire bands) in the VTX table, it will not always be possible to match a VTX table band to a predefined band.

Reply:
Maybe we could have the user explicity define the mapping:
`vtxtable band 5 RACEBAND R R1(5658) R2(5695) 5555 F6(5840) A3(5825) 5648 R6(5843)`
This would eliminate the need for the default frequency table. I would have to figure out a way to store both channel and frequency information though.

Reply:
This is getting quite complicated now.

What do you think about introducing a per band flag that allows the user to specify that this band is to be used as a 'factory band' and the frequency selected by sending the corresponding band / channel index to the VTX, with the frequencies for this band in the table becoming purely informational?

@jflyper, your thoughts?

Reply:
That sounds good, as long as it is in combination with empty placeholders. That way one could use index mode while still blocking illegal/unwanted frequencies. At the same time, unwanted factory bands could be overwritten with custom ones that use frequency mode.

The flexibility that my last suggestion would have provided was probably also overkill. After all, who wants a resorted raceband? Especially for interoperability it makes much more sense to just block unwanted frequencies but keep their positions the same.
In addition to the frequencies being in their stock positions, this would also require the bands to be in their stock positions. A small price to pay for the simplicity of this solution.

Reply:
Good point about empty placeholders - setting a frequency to 0 should work just right for this.

Re stock positions - not really, the user just needs to know what bands channels are where, and fill in the right (indicative) frequencies, or else they will end up with a display that does not match the actual frequency.

Reply:
Exactly. So users will neither be able to change the order of factory bands nor their frequencies. Custom bands will have to squeeze into otherwise unwanted spots or at the end. But as i said, small price to pay.

That leaves one question though: What about other vtx protocols? IRC Tramp does not seem to have a band/channel mode. The easiest solution would be to declare factory bands as invalid for tramp devices. After all, when the frequency is only informational, no real freq can be given to the vtx.
Or should the factory flags be part of the SmartAudio implementation and thus by design separate from vtxtable?

Reply:
I guess we'll need to at least support managing the 'factory band' flag at the higher VTX table levels. Protocols / drivers that do not support the concept of factory bands can then just completely skip doing anything when a user configures one and then tries to use it - part of the UX introduced by VTX tables is that it expects users to know what their hardware supports, and making it easy for them to do this by supporting the usage of pre-canned 'band plans'.